src/Sites/twitch.tv/Util/MessagePatcher.tsx

```diff
@@ -79,24 +83,35 @@ export class MessagePatcher {
 	/**
 	 * Render the message with 7TV emotes
 	 */
-	render(line: Twitch.ChatLineAndComponent): void {
+	render(line: Twitch.ChatLineAndComponent | Twitch.VideoMessageAndComponent): void {
 		// Hide twitch fragments
-		const oldFragments = Array.from(line.element.querySelectorAll<HTMLSpanElement | HTMLImageElement>('span.text-fragment, span.mention-fragment, a.link-fragment, img.chat-line__message--emote, [data-test-selector=emote-button]'));
+		// Note: Some VOD fragments, like Twitch emotes, contain an additional layout that also needs to be hidden. Hence: div[class^=InjectLayout-sc-]
+		// All user badges also have layouts, so they need to be excluded from removal.
+		const oldFragments = Array.from(line.element.querySelectorAll<HTMLSpanElement | HTMLImageElement>('span.text-fragment, span.mention-fragment, a.link-fragment, img.chat-line__message--emote, [data-test-selector=emote-button], div[class^=InjectLayout-sc-]'));
```

Comment:
```suggestion
		const oldFragments = Array.from(line.element.querySelectorAll<HTMLSpanElement | HTMLImageElement>('span.text-fragment, span.mention-fragment, a.link-fragment, img.chat-line__message--emote, div.chat-image__container, [data-test-selector=emote-button]'));
```

I believe it's because of this line, the wildcard class selector on `InjectLayout-sc-` also omits badges because they have this class. Perhaps this may work better?

Reply:
Yup. Sorry about that. Yeah, your method works a lot better. I'll also remove the exception I put in the old fragments loop, since that only worked for live chats anyway.

Reply:
Also to note 7TV badges still don't render, which I believe is because of line 67 in `MessageRenderer.tsx`. The selector there needs to be adapted

Reply:
I'll see what I can do. It seems badges aren't grouped together with the username in VODs like they are in live chats.